src/locale/localed.c

```diff
@@ -298,13 +299,59 @@ static int process_locale_list_item(
         return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Locale assignment %s not valid, refusing.", assignment);
 }
 
+static int locale_gen_process_locale(char *new_locale[static _VARIABLE_LC_MAX],
+                                     sd_bus_error *error) {
+        int r;
+        assert(new_locale);
+
+        for (LocaleVariable p = 0; p < _VARIABLE_LC_MAX; p++) {
+                if (p == VARIABLE_LANGUAGE)
+                        continue;
+                if (isempty(new_locale[p]))
+                        continue;
+                if (locale_is_installed(new_locale[p]))
+                        continue;
+
+                r = locale_gen_enable_locale(new_locale[p]);
+                if (r != 0) {
+                        if (r == -ENOEXEC) {
```

Comment:
why the duplicate checking of `r`? an attempt to optimize success paths?

let the compiler figure that out, and this isn't performance sensitive anyway.

we much prefer shallow indentation, i.e. deeply nesting `if` blocks unnecessarily should be avoided. hence, as mentioned do:

```c
r = …;
if (r == -ENOEXEC) {
        …
} else if (r == -EINVAL) {
        …
} else if (r < 0) {
        …
}
```

this way you only have one level of indentation. You handle the errors you want to handle explicitly first, then you handle all other errors, and the main code continues on the original indentation level.

(i.e. what i suggested earlier)